Subject: testparm: clarify "weak crypto is allowed" message
From: Michael Tokarev <mjt@tls.msk.ru>
Date: Fri, 01 Apr 2022 09:56:55 +0300
Bug-Debian: https://bugs.debian.org/975882
Bug: https://bugzilla.samba.org/show_bug.cgi?id=14583

This message makes people think there's some issue with their
smb.conf settings which allows weak crypto to be used while
communicating with (windows) clients.  This actually is not
the case, the message says weak algorithms are allowed by the
gnutls _library_, not by smb.conf. Clarify the message to
avoid confusion.

Signed-off-by: Michael Tokarev <mjt@tls.msk.ru>

diff --git a/source3/utils/testparm.c b/source3/utils/testparm.c
index 58ba46bc15f..4d419fd4805 100644
--- a/source3/utils/testparm.c
+++ b/source3/utils/testparm.c
@@ -875,7 +875,7 @@ static void do_per_share_checks(int s)
 	} else {
 		weak_crypo_str = "disallowed";
 	}
-	fprintf(stderr, "Weak crypto is %s\n", weak_crypo_str);
+	fprintf(stderr, "Weak crypto is %s by gnutls library\n", weak_crypo_str);
 
 	if (skip_logic_checks == 0) {
 		ret = do_global_checks();
